### PR TITLE
[O11y][Oracle WebLogic] Add dimension fields for deployed_application datastream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

This PR adds dimension fields for deployed_application datastream to support TSDB enablement.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

- Relates #7512

## Screenshot

Refer: https://github.com/elastic/integrations/issues/7512#:~:text=Screenshots-,Deployed%20Application,-%22oracle_weblogic%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%22properties%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22deployed_application

### CODE REVIEW

File: packages/oracle_weblogic/data_stream/deployed_application/fields/fields.yml [hunk 1]
```diff
@@ -40,6 +40,8 @@
         - name: source_info
           type: keyword
           description: Source info of the deployment as a keyword.
+          # Reason to add as a dimension field: There can be many different deployed applications source info in Oracle WebLogic instance.
+          dimension: true
         - name: status
```
Comment (agithomas) [inline packages/oracle_weblogic/data_stream/deployed_application/fields/fields.yml:45]: Is it possible to have a deployment to have duplicate name, one having Status "Running" and other "Stopped" ?

Comment (harnish-elastic) [reply packages/oracle_weblogic/data_stream/deployed_application/fields/fields.yml:45]: @agithomas, There is no possibility of deployment having a duplicate name.

**Review by agithomas — APPROVED** [review]
LGTM!

**Comment by agithomas** [comment]
Please complete the peer review additionally.

**Review by aliabbas-elastic — APPROVED** [review]

**Pull request merged**